### PR TITLE
Add reva-gateway endpoint to service discovery

### PR DESCRIPTION
**Pull request by IljaN**
Reva can't be wrapped as a go-micro service due to private apis. We workaround by manually registering the address and port in the service-registry.

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/external/external_test.go  9
- pkg/service/external/external.go  6
- pkg/command/gateway.go  1
         

Clones removed
==============
+ pkg/command/gateway.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=5336434&bid=17467801)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/external/external_test.go  10
- pkg/service/external/external.go  6
- pkg/command/gateway.go  1
         

Clones removed
==============
+ pkg/command/gateway.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=5336434&bid=17467801)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/external/external_test.go  10
- pkg/service/external/external.go  6
- pkg/command/gateway.go  1
         

Clones removed
==============
+ pkg/command/gateway.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=5336434&bid=17467801)

File: pkg/service/external/external_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import (
+	"context"
+	"testing"
+
+	"github.com/micro/go-micro/v2/registry"
+	"github.com/owncloud/ocis-pkg/v2/log"
+)
+
+func TestRegisterGRPCEndpoint(t *testing.T) {
```
Comment (refs) [inline pkg/service/external/external_test.go:11]: IMHO this test is not needed, this is an external dependency, you're testing micro.

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/external/external_test.go  10
- pkg/service/external/external.go  6
- pkg/command/gateway.go  1
         

Clones removed
==============
+ pkg/command/gateway.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=5336434&bid=17467801)

File: pkg/service/external/external_test.go (hunk 1 above)
Comment (IljaN) [reply pkg/service/external/external_test.go:11]: I would like to keep the test as: 
- it tests that the method really registers a service
- it tests the cancellation which removes the service.

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/external/external_test.go  10
- pkg/service/external/external.go  7
- pkg/command/gateway.go  1
         

Clones removed
==============
+ pkg/command/gateway.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=5336434&bid=17467801)

**Pull request merged**